staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/queueset.go

```diff
@@ -799,10 +821,10 @@ func (qs *queueSet) finishRequestLocked(r *request) {
 	releaseSeatsLocked := func() {
 		defer qs.removeQueueIfEmptyLocked(r)
 
-		qs.totSeatsInUse -= r.Seats()
-		metrics.AddRequestConcurrencyInUse(qs.qCfg.Name, r.fsName, -r.Seats())
+		qs.totSeatsInUse -= r.MaxSeats()
```

Comment:
if we define `FinalSeats` as the number of seats occupied at the end, during the `AdditionalLatency` then:
- `FinalSeats <= InitialSeats`:  we release `InitialSeats-FinalSeats` immediately after the request finishes and then hold `FinalSeats` for `AdditionalLatency`

- `FinalSeats > InitialSeats`: after the request finishes, we will need to acquire `FinalSeats - InitialSeats` more seats from the pool which is not a guaranteed process.

This PR now reserves `MaxSeats` when the request is dispatched, but this seems like an under utilization when `FinalSeats > InitialSeats`

Reply:
- Since `FinalSeats < InitialSeats` never actually happens, I would not worry about supporting it.
- Remember that when `InitialSeats < FinalSeats`, while this guarantees seats will not be occupied it does not guarantee under-utilization because our seats are just abstractions that are multiplexed onto the underlying real resources.  For example, when multiplexing fewer processes onto a fixed number of CPUs, those processes get work done faster (assuming CPU is the bottleneck and there are at least as many processes as CPUs).
- We have discussed the potential for under-utilization in review comments and meetings, and the current thinking is to let the first draft take no particular steps to combat it and see how that works out.